contrib/src/main/java/com/datatorrent/contrib/kafka/POJOKafkaOutputOperator.java

```diff
@@ -0,0 +1,272 @@
+import java.util.List;
+import java.util.Properties;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.lib.util.PojoUtils;
+
+import kafka.producer.KeyedMessage;
+import kafka.producer.ProducerConfig;
+
+/**
+ * POJOKafkaOutputOperator extends from AbstractKafkaOutputOperator receives the POJO
+ * from upstream and converts to Kafka Messages and writes to kafka topic.&nbsp;
+ * <p>
+ * <br>
+ * Ports:<br>
+ * <b>Input</b>: Have only one input port<br>
+ * <b>Output</b>: No Output Port <br>
+ * <br>
+ * Properties:<br>
+ * <b>brokerList</b>: List of brokers in the form of Host1:Port1,Host2:Port2,Host3:port3,...<br>
+ * <b>keyField</b>: Specifies the field creates distribution of tuples to kafka partition. <br>
+ * <b>isBatchProcessing</b>: Specifies whether to write messages in batch or not. By default,
+ *                           the value is true <br>
+ * <b>batchSize</b>: Specifies the batch size.<br>
+ * <br>
+ * <br>
+ * </p>
+ *
+ * @displayName POJO Kafka Output
+ * @category Messaging
+ * @tags Output operator
+ *
+ */
+public class POJOKafkaOutputOperator extends AbstractKafkaOutputOperator<Object,Object>
+{
+  @AutoMetric
+  private long outputMessagesPerSec;
+  @AutoMetric
+  private long outputBytesPerSec;
+  protected final Integer DEFAULT_BATCH_SIZE = 200;
+  protected final String BROKER_KEY = "metadata.broker.list";
+  protected final String BATCH_NUM_KEY = "batch.num.messages";
+  private long messageCount;
+  private long byteCount;
+  private String brokerList;
+  private double windowTimeSec;
+  private String keyField = "";
+  protected boolean isBatchProcessing = true;
+  @Min(2)
+  protected int batchSize = DEFAULT_BATCH_SIZE;
+  protected transient List<KeyedMessage<Object,Object>> messageList = new ArrayList<>();
+  protected transient PojoUtils.Getter keyMethod;
+  protected transient Class<?> pojoClass;
+  protected transient ProducerConfig producerConfig;
+
+  public final transient DefaultInputPort<Object> inputPort = new DefaultInputPort<Object>() {
+    @Override
+    public void setup(Context.PortContext context)
+    {
+      if(context.getAttributes().contains(Context.PortContext.TUPLE_CLASS)) {
+        pojoClass = context.getAttributes().get(Context.PortContext.TUPLE_CLASS);
+      }
+    }
+
+    @Override
+    public void process(Object tuple)
+    {
+      processTuple(tuple);
+    }
+  };
+
+  /**
+   * setup producer configuration.
+   * @return ProducerConfig
+   */
+  @Override
+  protected ProducerConfig createKafkaProducerConfig(){
+    if(brokerList != null) {
+      getConfigProperties().setProperty(BROKER_KEY, brokerList);
+    }
+    getConfigProperties().setProperty(BATCH_NUM_KEY, String.valueOf(batchSize));
```

Comment:
Yes. If it is a batch mode, added the producer.type as async to config properties.